docs/tutorials/oracle.md

```diff
@@ -0,0 +1,153 @@
+# Setting up ExternalDNS for Oracle Cloud Infrastructure (OCI)
+
+This tutorial describes how to setup ExternalDNS for usage within a Kubernetes cluster using OCI DNS.
+
+Make sure to use the latest version of ExternalDNS for this tutorial.
+
+## Creating an OCI DNS Zone
+
+Create a DNS zone which will contain the managed DNS records. Let's use `example.com` as an reference here.
+
+For more information about OCI DNS see the documentation [here][1].
+
+## Deploy ExternalDNS
+
+Connect your `kubectl` client to the cluster you want to test ExternalDNS with.
+We first need to create a config file containing the information needed to connect with the OCI API.
+
+Create a new file (oci.yaml) and modify the contents to match the example below. Be sure to adjust the values to match your own credentials:
+
+```yaml
+auth:
+  region: us-phoenix-1
+  tenancy: ocid1.tenancy.oc1...
+  user: ocid1.user.oc1...
+    -----BEGIN RSA PRIVATE KEY-----
+    -----END RSA PRIVATE KEY-----
+  fingerprint: af:81:71:8e...
+compartment: ocid1.compartment.oc1...
+```
+
+Create a secret using the config file above:
+
+```shell
+$ kubectl create secret generic external-dns-config --from-file=oci.yaml
+```
+
+### Manifest (for clusters with RBAC enabled)
+
+Apply the following manifest to deploy ExternalDNS.
+
+```yaml
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: external-dns
+---
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRole
+metadata:
+  name: external-dns
+rules:
+- apiGroups: [""]
+  resources: ["services"]
+  verbs: ["get","watch","list"]
+- apiGroups: [""]
+  resources: ["pods"]
+  verbs: ["get","watch","list"]
+- apiGroups: ["extensions"]
+  resources: ["ingresses"]
+  verbs: ["get","watch","list"]
```

Comment:
external-dns should be allowed to list nodes

- apiGroups: [""]
  resources: ["nodes"]
  verbs: ["list"]